aquadoggo/src/materializer/tasks/reduce.rs

```diff
@@ -17,15 +19,26 @@ use crate::materializer::TaskInput;
 /// After succesfully reducing and storing a document view an array of dependency tasks is returned.
 /// If invalid inputs were passed or a fatal db error occured a critical error is returned.
 pub async fn reduce_task(context: Context, input: TaskInput) -> TaskResult<TaskInput> {
+    debug!("Working on reduce task {:#?}", input);
+
     // Find out which document we are handling
-    let document_id = resolve_document_id(&context, &input).await?;
+    let document_id = match resolve_document_id(&context, &input).await? {
+        Some(document_id) => Ok(document_id),
+        None => {
+            debug!("No document found for this view, exit without dispatching any other tasks");
+            return Ok(None);
+        }
+    }?;
```

Comment:
Instead of returning `Ok(None)` when a document for a requested view didn't exist, we were critically failing the task here.

Logic bug....